### ISSUE TITLE
apple_common may only be used from one of the following repositories or prefixes error in rules_swift

### ISSUE THREAD

**Issue by brentleyjones** [bazelbuild/bazel#24375]
### Description of the bug:

https://buildkite.com/bazel/rules-swift-swift/builds/5879#01934108-e472-404d-9ee5-8bf41061f175/1476-1494

```
(20:49:14) ERROR: /workdir/swift/swift_compiler_plugin.bzl:303:51: apple_common may only be used from one of the following repositories or prefixes: @@_builtins//, @@apple_support//, @@bazel_tools//, @@local_config_cc//, @@rules_apple//, @@rules_cc//, @@rules_go//, @@rules_ios//, @@rules_swift//, @@stardoc//, @@tulsi//, test_starlark, tools/osx
```

But this is rules_swift. Is it because we have `repo_name = "build_bazel_rules_swift"` set in `MODULE.bazel`? If so, that's should be supported.

### What is the output of `bazel info release`?

89c634ff601a367b9f55a90868f13c35fbe57d61

**Comment by brentleyjones** [comment]
Because of https://github.com/bazelbuild/bazel/commit/b406bdcbe95dd23ca623f807d5113f18b724611a.

**Comment by fmeum** [comment]
@pzembrod

**Comment by brentleyjones** [comment]
@bazel-io flag

**Comment by iancha1992** [comment]
@bazel-io fork 8.0.0

**Comment by Wyverald** [comment]
This is breaking rules_android too (because of rules_go). cc @ahumesky @comius

**Comment by ahumesky** [comment]
Yes also seeing this from rules_android with 8.0.0 release branch at 62493bda06cc1db3d4fb03faa8ac96b42b70820a:

`ERROR: .../.cache/bazel/_bazel_ahumesky/cb1fcf0b7315e9ad5d7a5ab005ad9fdb/external/io_bazel_rules_go/go/platform/apple.bzl:16:25: apple_common may only be used from one of the following repositories or prefixes: @@_builtins//, @@apple_support//, @@bazel_tools//, @@local_config_cc//, @@rules_apple//, @@rules_cc//, @@rules_go//, @@rules_ios//, @@rules_swift//, @@stardoc//, @@tulsi//, test_starlark, tools/osx`

rules_go refers to itself as io_bazel_rules_go e.g. here:
[https://github.com/bazel-contrib/rules_go/blob/63b0dcb1ebf8678997fea7a6b253214aa4a9f447/go/private/go_toolchain.bzl#L210-L222](https://www.google.com/url?sa=D&q=https%3A%2F%2Fgithub.com%2Fbazel-contrib%2Frules_go%2Fblob%2F63b0dcb1ebf8678997fea7a6b253214aa4a9f447%2Fgo%2Fprivate%2Fgo_toolchain.bzl%23L210-L222)

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 8.0.0 RC3](https://github.com/bazelbuild/bazel/releases/tag/8.0.0rc3). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=8.0.0rc3. Thanks!
